frontend/src/hooks/useAddRedactionToState.jsx

```diff
@@ -0,0 +1,39 @@
+import {
+    redactionIDsState,
+    redactionsState,
+  } from "@/util/CommentRedactionState";
+  
+  import { useRecoilCallback } from "recoil";
+  
+  //call this to add a new redaction
+  export default function useAddRedactionToState() {
+    console.log(redactionIDsState)
+    return useRecoilCallback(
+      ({ set }) => (id, threadData) => {
+        set(redactionIDsState, (ids) => new Set([...Array.from(ids), id]));
+        set(redactionsState(id), threadData);
+      },
+      []
+    );
+  }
+
+  export default function useRemoveRedactionFromState(mark) {
```

Comment:
this function doesn't work. i added a lot of print statements to try & debug but getting an index of error